prac_07/project_management.py

```diff
@@ -0,0 +1,199 @@
+"""
+CP1404 Practical - Project Management Program - Client Side.
+Estimated completion time: 2.5 hours
+Actual completion time: 8 hours
+"""
+import datetime
+from operator import attrgetter
+from project import Project
+
+MENU = "- (L)oad projects\n- (S)ave projects\n- (D)isplay projects\n- (F)ilter projects by date\n" \
+       "- (A)dd new project\n- (U)pdate project\n- (Q)uit"
+FILENAME = "projects.txt"
+
+
+def main():
+    """Project Management Program."""
+    projects = []
+    load_projects(FILENAME, projects)
+    print(MENU)
+    menu_choice = input(">>> ").upper()
+    while menu_choice != "Q":
+        if menu_choice == "L":
+            projects = load_new_file()
+        elif menu_choice == "S":
+            new_save_file = input("Enter new save file: ")
+            save_projects(new_save_file, projects)
+        elif menu_choice == "D":
+            display_projects_status(projects)
+        elif menu_choice == "F":
+            filter_projects(projects)
+        elif menu_choice == "A":
+            print("Let's add a new project")
+            add_new_project(projects)
+        elif menu_choice == "U":
+            display_projects(projects)
+            update_project(projects)
+        else:
+            print("Invalid option")
+        print(MENU)
+        menu_choice = input(">>> ").upper()
+    save_projects(FILENAME, projects)
+    print("Thank you for using custom-built project management software.")
+
+
+def load_new_file():
+    """Gets a new valid file to load."""
+    new_projects = []
+    invalid_input = True
+    while invalid_input:
+        try:
+            new_project_file = input("Enter file you wish to load: ")
+            load_projects(new_project_file, new_projects)
+            invalid_input = False
+        except FileNotFoundError:
+            print(f"Unable to find file {new_project_file}")
+    return new_projects
+
+
+def filter_projects(projects):
+    """Displays projects from a specified date range."""
+    filter_date = get_valid_date("Show projects that start after date (dd/mm/yyyy): ")
```

Comment:
I think filter_date is a bit misleading for variable name as it sounds like verb. But, understandable for context.